autoscale/src/main/java/com/sequenceiq/periscope/service/NodeDeletionService.java

```diff
@@ -0,0 +1,48 @@
+package com.sequenceiq.periscope.service;
+
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackV4Response;
+import com.sequenceiq.periscope.domain.Cluster;
+import com.sequenceiq.periscope.domain.LoadAlert;
+import com.sequenceiq.periscope.monitor.handler.CloudbreakCommunicator;
+import com.sequenceiq.periscope.utils.StackResponseUtils;
+
+@Service
+public class NodeDeletionService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NodeDeletionService.class);
+
+    @Inject
+    private CloudbreakCommunicator cloudbreakCommunicator;
+
+    @Inject
+    private StackResponseUtils stackResponseUtils;
+
+    public void deleteStoppedNodesIfPresent(Cluster cluster) {
+        Set<LoadAlert> loadAlerts = cluster.getLoadAlerts();
+        if (loadAlerts.isEmpty()) {
+            LOGGER.info("No load-based policy present for cluster: {}, skipping request to delete stopped nodes", cluster.getStackCrn());
+            return;
+        }
+
+        StackV4Response stackResponse = cloudbreakCommunicator.getByCrn(cluster.getStackCrn());
+        LoadAlert loadAlert = loadAlerts.iterator().next();
+        String hostGroup = loadAlert.getScalingPolicy().getHostGroup();
+        List<String> stoppedHostIds = stackResponseUtils.getStoppedCloudInstanceIdsInHostGroup(stackResponse, hostGroup);
+
+        if (stoppedHostIds.isEmpty()) {
+            LOGGER.info("No stopped nodes found on cluster: {}, skipping request to delete stopped nodes", cluster.getStackCrn());
+        } else {
+            LOGGER.info("Sending request to delete stopped nodes with hostIds: {} for cluster: {}", stoppedHostIds, cluster.getStackCrn());
+            cloudbreakCommunicator.deleteInstancesForCluster(cluster, stoppedHostIds);
```

Comment:
What happens if there's a current Cloudbreak operation in progress?  This one will be particularly tricky to handle. If there's no policies for a cluster - scaling will be disabled, and there'll be no mechanism to check for the state.

That said, adding another Service which monitors a cluster for Stopped nodes has a few issues:
1) We may not have caused the Stopped instances.
2) We won't know the hostGroup - since there's no autoscaling policy any more.

Could be resolved by either 1) Storing this as a pending operation on the cluster (in case it fails), or 2) Best-effort trying to delete nodes in such scenarios.

Reply:
> 2) Best-effort trying to delete nodes in such scenarios.

I think this is the better approach - delete only if applicable. In case the cluster has an operation in progress, node deletion is skipped entirely. 

Approach 1 would involve editing the cluster entity by introducing an additional column which would store info about node deletion.

Reply:
Updated this with a best-effort style implementation with corresponding failure messages being displayed on the activity console